cellxgene_resources/curation_sample_code.ipynb

```diff
@@ -651,6 +757,92 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Extracts gene_id and gene_symbol from gtf\n",
+    "\n",
+    "def extract_gene_id_and_symb(dir, gtf):\n",
+    "    '''\n",
+    "    Input: path to gtf file\n",
+    "    Output: dataframe of the gene_ids and gene_symbols\n",
+    "    '''\n",
+    "    gtf = pd.read_table(dir + gtf, header=None, comment = '#')\n",
+    "    row_8 = gtf[8]\n",
+    "    pattern1 = ';'\n",
+    "    pattern2 = '(ENSG\\d{11})'\n",
+    "    pattern3 = '(gene_name)'\n",
+    "    pattern4 = 'gene_name'\n",
+    "    pattern5 = '([\\w\\d\\-]+)'\n",
+    "    split1 = row_8.str.split(pattern1, expand=True).dropna(axis=1)\n",
+    "    gene_id_col = split1.loc[:,split1.iloc[0].str.contains(pattern2)].columns\n",
+    "    gene_sym_col = split1.loc[:,split1.iloc[0].str.contains(pattern3)].columns\n",
+    "    gene_ids = split1[gene_id_col].iloc[:,0].str.split(pattern2, expand=True)[1]\n",
+    "    gene_sym = split1[gene_sym_col].iloc[:,0].str.split(pattern4, expand=True)[1].str.split(pattern5, expand=True)[1]\n",
+    "    gtf_df = pd.DataFrame({'gene_ids':gene_ids, 'gene_symbols':gene_sym})\n",
+    "    gtf_df.dropna(axis=0, inplace=True)\n",
+    "    gtf_df.set_index('gene_ids')\n",
+    "    return gtf_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Filters out gene_symbols with multiple gene_ids and assigns 'multiple' as gene_id to corresponding gene_symbol\n",
+    "def assign_multiple(gtf_df):\n",
+    "    '''\n",
+    "    Input: gtf pandas dataframe with gene_id and gene_symbol columns\n",
+    "    Output: de-duplicated dataframe\n",
+    "    '''\n",
+    "    dups = gtf_df[gtf_df.duplicated(subset='gene_symbols',keep=False)]\n",
+    "    dups['gene_ids'] = 'multiple'\n",
+    "    dups = dups.drop_duplicates()  # only keep one instance of the gene_symbol\n",
+    "    gene_df_wo_dups = gene_df.drop_duplicates(subset='gene_symbols',keep=False) \n",
+    "    gene_df_multiple = gene_df_wo_dups.append(dups)  #create new df with all non-duplicated gene_symbols and gene_symbols with multiple ensembl ids\n",
+    "    gene_df_multiple.set_index('gene_ids', inplace=True)\n",
+    "    return gene_df_multiple"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# can include multiple gtf files (list of gtf files)\n",
+    "# gtf_list = ['gtf','gtf']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Extract and de-duplicate gene annotations and write them to a tsv file\n",
+    "for file in gtf_list:\n",
+    "    gtf_df = extract_gene_id_and_symb(file)\n",
+    "    gene_id_symbol = assign_multiple(gtf_df)\n",
+    "\n",
+    "    # write dataframe to tsv file in ref_file folder\n",
+    "    dir = '/path/to/cxg_curation/ref_files/'\n",
```

Comment:
I think separating out `dir` from the file name is unnecessary as I don't see any use for it to be used independently
So I would simplify things to just accept the full path in a single variable (which some notebooks are set up to autocomplete easily anyway).